records/generator_test.go

```diff
@@ -13,6 +14,129 @@ func init() {
 	logging.SetupLogs()
 }
 
+func TestMasterRecord(t *testing.T) {
+	// masterRecord(domain string, masters []string, leader string)
+	type expectedRR struct {
+		name  string
+		host  string
+		rtype string
+	}
+	tt := []struct {
+		domain  string
+		masters []string
+		leader  string
+		exists  []expectedRR
+	}{
+		{"foo.com", nil, "", nil},
+		{"foo.com", nil, "@", nil},
+		{"foo.com", nil, "1@", nil},
+		{"foo.com", nil, "@2", nil},
+		{"foo.com", nil, "3@4", nil},
+		{"foo.com", nil, "5@6:7",
+			[]expectedRR{
+				{"leader.foo.com.", "6", "A"},
+				{"master.foo.com.", "6", "A"},
+				{"master0.foo.com.", "6", "A"},
+				{"_leader._tcp.foo.com.", "leader.foo.com.:7", "SRV"},
+				{"_leader._udp.foo.com.", "leader.foo.com.:7", "SRV"},
+			}},
+		{"foo.com", []string{"6:7"}, "5@6:7",
+			[]expectedRR{
+				{"leader.foo.com.", "6", "A"},
+				{"master.foo.com.", "6", "A"},
+				{"master0.foo.com.", "6", "A"},
+				{"_leader._tcp.foo.com.", "leader.foo.com.:7", "SRV"},
+				{"_leader._udp.foo.com.", "leader.foo.com.:7", "SRV"},
+			}},
+		{"foo.com", []string{"8:9"}, "5@6:7",
+			[]expectedRR{
+				{"leader.foo.com.", "6", "A"},
+				{"master.foo.com.", "6", "A"},
+				{"master.foo.com.", "8", "A"},
+				{"master1.foo.com.", "6", "A"},
+				{"master0.foo.com.", "8", "A"},
+				{"_leader._tcp.foo.com.", "leader.foo.com.:7", "SRV"},
+				{"_leader._udp.foo.com.", "leader.foo.com.:7", "SRV"},
+			}},
+		{"foo.com", []string{"8:9", "8:9"}, "5@6:7",
+			[]expectedRR{
+				{"leader.foo.com.", "6", "A"},
+				{"master.foo.com.", "6", "A"},
+				{"master.foo.com.", "8", "A"},
+				{"master1.foo.com.", "6", "A"},
+				{"master0.foo.com.", "8", "A"},
+				{"_leader._tcp.foo.com.", "leader.foo.com.:7", "SRV"},
+				{"_leader._udp.foo.com.", "leader.foo.com.:7", "SRV"},
+			}},
+		{"foo.com", []string{"8:9", "6:7"}, "5@6:7",
```

Comment:
Can we add short descriptive comments explaining what each of these tests cases is verifying (where it makes sense)?